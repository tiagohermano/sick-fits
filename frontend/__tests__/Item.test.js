import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: '123',
  title: 'Test Item',
  price: 1000,
  description: 'This is a test item',
  image: 'testItem.jpg',
  largeImage: 'testItem-large.jpg',
}

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders the title and pricetag properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    const priceTag = wrapper.find('PriceTag');
    expect(priceTag.children().text()).toBe('$10');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
})