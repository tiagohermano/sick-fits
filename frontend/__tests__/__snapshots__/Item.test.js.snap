// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Test Item"
    src="testItem.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        Test Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $10
  </PriceTag>
  <p>
    This is a test item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="123"
    />
    <DeleteItem
      id="123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
